my 3rd homework

MogulKhan93 opened the pull request:
ver. 1.0

WiWhite commented on Baliuk/homework3/file3.py:
```diff
@@ -0,0 +1,74 @@
+users_data_dict = {}
+
+for i in range(l):
+    users_data_dict.update({users_data[i].split()[0]: users_data[i].split()[1]})  # creating a dictionary with users data
```
эта строка вызывает исключение, когда в систему пытаеться зарегистрироватся второй пользователь

MogulKhan93 commented:
Привіт
Трішки не зрозумів..(
В цьому рядку я беру дані з файла (логін і пароль вже зареєстрованих
користувачів), щоб перевести їх і створити словник, з яким буду далі
оперувати

чт, 3 черв. 2021, 11:26 користувач WiWhite ***@***.***> пише:

> *@WiWhite* commented on this pull request.
> ------------------------------
>
> In Baliuk/homework3/file3.py
> <https://github.com/WiWhite/home_works/pull/52#discussion_r644588367>:
>
> > +# 5. Если желает, взять от него необходимые данные и вывести об успешной
> +# регистрации, если не желает регистрироватся - пожелать удачи.
> +# 6. Данные о зарегестрированных пользователях хранить в файле 'users.txt',
> +# по желанию можете создать файл для логирования событий регистрации и входа.
> +
> +regist_quest = input('Hello. Have you previously registered on our resource? \
> +                      \nif \"yes\" - enter \"1\" \nif \"no\" - enter \"2\" ')
> +
> +with open('users.txt', 'r') as file:
> +    users_data = file.readlines()
> +
> +l = len(users_data)  # quantity of registered users
> +users_data_dict = {}
> +
> +for i in range(l):
> +    users_data_dict.update({users_data[i].split()[0]: users_data[i].split()[1]})  # creating a dictionary with users data
>
> эта строка вызывает исключение, когда в систему пытаеться
> зарегистрироватся второй пользователь
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/WiWhite/home_works/pull/52#pullrequestreview-675007226>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4OKEU52X2T6IX7EWY3DRTTQ44EXANCNFSM457TCGOQ>
> .
>

MogulKhan93 commented:
Саш, підскажи, що саме треба переробити?
Я довго тестував..вроді все добре))

чт, 3 черв. 2021, 11:32 користувач Тольбашка ***@***.***> пише:

> Привіт
> Трішки не зрозумів..(
> В цьому рядку я беру дані з файла (логін і пароль вже зареєстрованих
> користувачів), щоб перевести їх і створити словник, з яким буду далі
> оперувати
>
> чт, 3 черв. 2021, 11:26 користувач WiWhite ***@***.***>
> пише:
>
>> *@WiWhite* commented on this pull request.
>> ------------------------------
>>
>> In Baliuk/homework3/file3.py
>> <https://github.com/WiWhite/home_works/pull/52#discussion_r644588367>:
>>
>> > +# 5. Если желает, взять от него необходимые данные и вывести об успешной
>> +# регистрации, если не желает регистрироватся - пожелать удачи.
>> +# 6. Данные о зарегестрированных пользователях хранить в файле 'users.txt',
>> +# по желанию можете создать файл для логирования событий регистрации и входа.
>> +
>> +regist_quest = input('Hello. Have you previously registered on our resource? \
>> +                      \nif \"yes\" - enter \"1\" \nif \"no\" - enter \"2\" ')
>> +
>> +with open('users.txt', 'r') as file:
>> +    users_data = file.readlines()
>> +
>> +l = len(users_data)  # quantity of registered users
>> +users_data_dict = {}
>> +
>> +for i in range(l):
>> +    users_data_dict.update({users_data[i].split()[0]: users_data[i].split()[1]})  # creating a dictionary with users data
>>
>> эта строка вызывает исключение, когда в систему пытаеться
>> зарегистрироватся второй пользователь
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/WiWhite/home_works/pull/52#pullrequestreview-675007226>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AT4OKEU52X2T6IX7EWY3DRTTQ44EXANCNFSM457TCGOQ>
>> .
>>
>

WiWhite commented on Baliuk/homework3/file3.py:
```diff
@@ -0,0 +1,74 @@
+regist_quest = input('Hello. Have you previously registered on our resource? \
+                      \nif \"yes\" - enter \"1\" \nif \"no\" - enter \"2\" ')
+
+with open('users.txt', 'r') as file:
```
если такого файла не будет я получу ошибку, а нужно чтобы программа создала файл в случае его отсутствия) сделай и попробуй потестить все снова

MogulKhan93 commented:
Аа, я зрозумів
Ок, дякую

чт, 3 черв. 2021, 14:16 користувач WiWhite ***@***.***> пише:

> *@WiWhite* commented on this pull request.
> ------------------------------
>
> In Baliuk/homework3/file3.py
> <https://github.com/WiWhite/home_works/pull/52#discussion_r644706047>:
>
> > +# 2. При запуске программы, пользователя должно спросить проходил ли он
> +# регистрацию на нашем ресурсе, если да, тогда предложить ему ввести логин и
> +# пароль от его учетной записи.
> +# 3. Если данные верны вывести сообщение об успешном входе в систему, если
> +# нет - тогда сообщить об это.
> +# 4. Если пользователь не регистрировался на ресурсе, тогда спросить не желает
> +# ли он пройти регистрацию.
> +# 5. Если желает, взять от него необходимые данные и вывести об успешной
> +# регистрации, если не желает регистрироватся - пожелать удачи.
> +# 6. Данные о зарегестрированных пользователях хранить в файле 'users.txt',
> +# по желанию можете создать файл для логирования событий регистрации и входа.
> +
> +regist_quest = input('Hello. Have you previously registered on our resource? \
> +                      \nif \"yes\" - enter \"1\" \nif \"no\" - enter \"2\" ')
> +
> +with open('users.txt', 'r') as file:
>
> если такого файла не будет я получу ошибку, а нужно чтобы программа
> создала файл в случае его отсутствия) сделай и попробуй потестить все снова
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/WiWhite/home_works/pull/52#pullrequestreview-675162382>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT4OKEVMQVWA6JP74VN7HLDTQ5QARANCNFSM457TCGOQ>
> .
>
